### DISCUSSION TITLE
Understanding the loader

### DISCUSSION THREAD

**Discussion by ajthinking** [excaliburjs/Excalibur#2491] [Q&A]
Hi community 👋 
I can't get the loader to work using this simple code:

```ts
import { Engine, ImageSource, Loader } from "excalibur";

const zombie = new ImageSource("zombie1.png");

const game = new Engine({});
const loader = new Loader([zombie]);
game.start(loader);

```

When using `yarn start`, the loader freezes:

<img width="713" alt="image" src="https://user-images.githubusercontent.com/3457668/189520915-41cd6218-cb77-4d1f-b140-900be4c33f42.png">

`zombie1.png` is in the same dir as `main.ts`:
![zombie1](https://user-images.githubusercontent.com/3457668/189520959-7442f626-6611-49b4-9db7-b852d347579a.png)

No errors in console. Network tab shows a 200 for the png. However - changing the filename to something that *does not exist* also gives 200 (!)

What am I missing here?  it would be nice if it could throw some error to help me.

**Comment by ajthinking** [comment]
Found a working example. Changing code to this works:

```ts
import { Engine, ImageSource, Loader } from "excalibur";

const zombieImage = require("./zombie1.png");
const zombie = new ImageSource(zombieImage);

const game = new Engine({});
const loader = new Loader([zombie]);
game.start(loader);
```

Im surprised by this as `ImageSource` expects: 

> @param path — Path to the image resource relative from the HTML document hosting the game, or absolute.

But path does not work, only a *required* image?

**Comment by chrisk-7777** [comment]
Hmm, it's odd that the ImageSource loader isn't working. I did a few tests locally just now and it looks ok. 
Here is a code pen: https://codepen.io/chrisk7777/pen/mdLERGY?editors=0010

The fact that `require` works makes me wonder if something bundling-related is going on? Thats just a wild guess in the dark though! Can you tell me a bit more about the stack you are using, any bundlers (like vite, etc)

Regarding the 404, if I try to load a missing image, I receive this: 

![image](https://user-images.githubusercontent.com/55621012/189524445-5976159f-8c31-4da0-bc3a-ab53083bc002.png)

If you try to visit `http://localhost:8000/not-an-image.png` (or whatever your local path is), does your local server return a 404?

**Comment by ajthinking** [comment]
I forked the breakout demo. Using yarn and chrome.
Here is a repo with my changes: https://github.com/ajthinking/sample-breakout

**Comment by ajthinking** [comment]
Visiting `http://localhost:8000/not-an-image.png` gives:

<img width="751" alt="image" src="https://user-images.githubusercontent.com/3457668/189525103-b0330977-bd9a-4777-9efb-5bb2adf8fbad.png">

**Comment by chrisk-7777** [comment]
Thanks @ajthinking I can see what is happening now. 
I believe your image (and other assets) should be placed in the static folder. Parcel will be looking for those there (https://github.com/ajthinking/sample-breakout/blob/main/package.json#L22)

**Comment by ajthinking** [comment]
Makes sense. Could not get that to work though: https://github.com/ajthinking/sample-breakout/commit/4a0ffd5f375b0147bb026083cefd44dfb539836b

**Comment by chrisk-7777** [comment]
Hmm, yea something definitely isn't right there. I'm not super familiar with Parcel, but I cloned down your repo and tested locally and the parcel server just isn't serving the assets (so that is outside of excalibur)

Personally I use Vite or Webpack, which might be better at serving up actual files. 
The next Excalibur sample is this one, which is using webpack and has external assets: https://github.com/excaliburjs/sample-platformer. Perhaps its better to swap to that?

**Comment by ajthinking** [comment]
The docs recommended parcel but I don't really know anything about it so switching to webpack might be a good idea. Thanks!
